src/schoology-extractor/schoology_extractor/api/request_client.py

```diff
@@ -3,21 +3,32 @@
 # The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
 # See the LICENSE and NOTICES files in the project root for more information.
 
+from dataclasses import dataclass
+import os
 import time
 import random
 from typing import List
 
+from opnieuw import retry
+from requests.exceptions import ConnectionError, HTTPError, Timeout
+from requests.packages.urllib3.exceptions import ProtocolError  # type: ignore
 from requests_oauthlib import OAuth1Session  # type: ignore
 
 from .paginated_result import PaginatedResult
 
 
-DEFAULT_URL = "https://api.schoology.com/v1/"
+DEFAULT_URL = os.environ.get("SCHOOLOGY_BASE_URL") or "https://api.schoology.com/v1/"
```

Comment:
Is this a value that we can pass as an arg/.env? or it is only going to be configurable from .env?

Reply:
Same question for REQUEST_RETRY_COUNT and REQUEST_RETRY_TIMEOUT_SECONDS

Reply:
I see these as changing rarely, so I didn't want to make them into full input parameters - especially the URL. But I did want a way to introduce different values in rare circumstances.

Reply:
I'd at least include those values in the `.env.example` or in the `readme` file.

Reply:
The RETRY ones should be included and I'll add that to the readme. The base url I would rather keep hidden for now.

Reply:
Done. And more 😆.